Link to repository table in documentation repository (README change only)

Ein-Tim opened the pull request:
## Checklist

* [X] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [X] Make sure `mvn -P integration-tests clean verify` runs for the whole project and, if you touched any code for a service in the `services` folder, ensure it can be run with `spring-boot:run`

## Description

This PR discontinues the [repository table](https://github.com/corona-warn-app/cwa-server#repositories) in the README here, instead the README now links to the [table](https://github.com/corona-warn-app/cwa-documentation/blob/master/README.md#repositories) in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation).
This will make it easier to update the list with newly created repositories.

Ein-Tim committed:
Update repositories table

This commit updates the repositories table.

hilmarf commented on README.md:
```diff
@@ -206,29 +206,33 @@ The German government has asked SAP and Deutsche Telekom to develop the Corona-W
+| [cwa-wishlist]            | Community feature requests.                                                  |
+| [cwa-website]             | The official website for the Corona-Warn-App.                                |
+| [cwa-server]              | Backend implementation for the Apple/Google exposure notification API.       |
+| [cwa-ppa-server]          | Backend implementation for the privacy-preserving analytics server.          |
```
```suggestion
| [cwa-ppa-server]          | Backend implementation for the data donation server.          |
```

hilmarf committed:
Update README.md

Ein-Tim commented:
I'll update this PR like https://github.com/corona-warn-app/cwa-ppa-server/pull/169

Ein-Tim committed:
Discontinue table of repositories in this repository

This commit discontinues the table of repositories, instead it links to the table in the documentation repository.

hilmarf committed:
Merge branch 'main' into fix/repo-table-update

hilmarf committed:
Merge branch 'main' into fix/repo-table-update
